.github/actions/docker-build-push/action.yml

```diff
@@ -0,0 +1,86 @@
+name: Build and Push Docker Image
+description: "Build and push a Docker image"
+inputs:
+  environment:
+    description: Environment (network) for workflow execution
+    required: false
+  imageName:
+    description: Name for the image
+    required: true
+  context:
+    description: Context for docker build execution
+    required: true
+  push:
+    description: True if the image should be published
+    required: true
+    default: "false"
+  gcrJsonKey:
+    description: JSON key for Google Container Registry service account (required if push is true)
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+
+    - name: Cache Docker layers
+      uses: actions/cache@v3
+      with:
+        path: /tmp/.buildx-cache
+        key: ${{ runner.os }}-buildx-${{ github.sha }}
+        restore-keys: |
+          ${{ runner.os }}-buildx-
+
+    - name: Load environment variables
+      if: env.PUSH == 'true'
```

Comment:
How does the value of `PUSH` environment variable get set? Same question for other variables like `ENVIRONMENT`, `GCR_JSON_KEY`, `CONTEXT`?

Reply:
It was set automatically with inputs. But you are right, it's not clear, I updated it with f1d3a6a4f002b8a2301ec47eefdd3d31b027deac